src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -203,9 +212,26 @@ namespace
             || code == ErrorCodes::CANNOT_READ_ALL_DATA
             || code == ErrorCodes::UNKNOWN_CODEC
             || code == ErrorCodes::CANNOT_DECOMPRESS
+            || code == ErrorCodes::DISTRIBUTED_BROKEN_BATCH_INFO
+            || code == ErrorCodes::DISTRIBUTED_BROKEN_BATCH_FILES
             || (!remote_error && code == ErrorCodes::ATTEMPT_TO_READ_AFTER_EOF);
     }
 
+    bool isSplittableErrorCode(int code, bool remote)
+    {
```

Comment:
Splittable error? :) sounds weird... 

Maybe smth like 'is error retrieable with smaller batch'  or 'error may relate to big batch size'?

Reply:
Added a comment.